### ISSUE TITLE
Enhance A11y Audit with GitHub Checks

### ISSUE THREAD

**Issue by seanforyou23** [patternfly/patternfly-next#649]
GitHub's new Checks API offers a better way for contributors to review our accessibility audit results  on github. We currently have status indicators for pass/fail/progress scenario, which is great, but it requires users to dig through lots of build logs to find the relevant results and take action.

By integrating with GitHub Checks, we'll have a mechanism in place where contributors can go to review and rerun reports for various code metrics like css/js linting, code coverage, a11y audit(s), etc. This makes for an awesome developer experience, which is critical to have in place as we expect to attract more and more outside contributors to the project.

For the accessibility audit specifically, we'd like to be able to split up the current test into multiple tests so we can target for example Section508 vs WCAG2A, or color contrast only for the demo app, or color contrast for only the components, etc. - where each variation of the audit can correspond to a different tab where the specific test results can be viewed.

Below is an example of where this type of report can be viewed from and what it could look like.

![missing-a11y-checks](https://user-images.githubusercontent.com/5942899/44694548-1ad92c00-aa3c-11e8-9549-a98cfa9883d3.png)

Another example with various "Checks" exposed.

<img width="603" alt="screen shot 2018-08-27 at 8 58 40 pm" src="https://user-images.githubusercontent.com/5942899/44694569-32181980-aa3c-11e8-8b2f-9162aa20b44f.png">

This is follow up work from https://github.com/patternfly/patternfly-next/issues/522. See links below for more information on what's possible

- https://blog.github.com/2018-05-07-introducing-checks-api/
- https://blog.travis-ci.com/2018-05-07-announcing-support-for-github-checks-api-on-travis-ci-com
- https://developer.github.com/changes/2018-05-07-new-checks-api-public-beta/
